app/model/commands/ExpiryFieldPreviewCommand.scala

```diff
@@ -0,0 +1,67 @@
+package model.commands
+
+import java.util.Date
+
+import com.gu.contentatom.thrift.{ContentAtomEvent, EventType}
+import com.gu.media.logging.Logging
+import com.gu.media.model.{ChangeRecord, MediaAtom}
+import com.gu.media.util.MediaAtomImplicits
+import com.gu.pandomainauth.model.{User => PandaUser}
+import data.DataStores
+import model.commands.CommandExceptions._
+import org.joda.time.DateTime
+
+import scala.util.{Failure, Success}
+
+case class ExpiryFieldPreviewCommand(override val stores: DataStores, user: PandaUser)
+  extends Command
+    with MediaAtomImplicits
+    with Logging {
+
+  type T = List[MediaAtom]
+
+
+  def process(): T = {
+    previewDataStore.listAtoms.fold(
+      {
+        err => AtomUpdateFailed(err.msg)
+      },
+
+      {
+        atoms => atoms.map(atom => reIndex(MediaAtom.fromThrift(atom)))
+      }
+    )
+  }
+
+  def reIndex(atom: MediaAtom) = {
+    log.info(s"Request to update atom ${atom.id}")
+    val expiry: Option[DateTime] = atom.expiryDate.map(expiry => new DateTime(expiry))
+
+    val details = atom.contentChangeDetails.copy(
```

Comment:
We're updating all atoms here, I think it'll be better if we only update the ones that have the old field set:


```scala
expiry match {
  case expDate: Some(DateTime) => {
    // update store and send message
  }
  case None => {
    // don't do anything, just return the same atom
    atom
  }
}

```